homeassistant/components/yale/manifest.json

```diff
@@ -1,8 +1,10 @@
 {
   "domain": "yale",
   "name": "Yale",
+  "after_dependencies": ["cloud"],
```

Comment:
It's not an after dependency since it's always required.

Reply:
AssertionError: No mock registered for GET https://account-link.nabucasa.com/services None

Do you know if we have a mock for account link?

Reply:
I guess I could copy this and adapt it https://github.com/home-assistant/core/blob/ada6b7875c2a7f4a54ba9dd93c70a93047d71874/tests/components/cloud/test_account_link.py#L147 but that kinda feels wrong to patch `hass_nabucasa` in this integration

Reply:
Normally we patch the (function) name in the dependency integration that is used by the integration in question.

Reply:
https://account-link.nabucasa.com/services

```
[
  {
    "service": "yale",
    "min_version": "2024.1.0",
    "accepts_new_authorizations": true
  },
  {
    "service": "xbox",
    "min_version": "0.116.0",
    "accepts_new_authorizations": true
  },
  {
    "service": "electric_kiwi",
    "min_version": "2023.2.0",
    "accepts_new_authorizations": true
  },
  {
    "service": "geocaching",
    "min_version": "2022.5.0",
    "accepts_new_authorizations": true
  },
  {
    "service": "seitron",
    "min_version": "2023.1.0",
    "accepts_new_authorizations": true
  },
  {
    "service": "lametric",
    "min_version": "2022.9.0",
    "accepts_new_authorizations": true
  },
  {
    "service": "spotify",
    "min_version": "0.116.0",
    "accepts_new_authorizations": false
  },
  {
    "service": "aladdin_connect",
    "min_version": "2024.6.0",
    "accepts_new_authorizations": true
  },
  {
    "service": "yolink",
    "min_version": "2022.5.0",
    "accepts_new_authorizations": true
  },
  {
    "service": "aladdin_genie",
    "min_version": "2024.1.0",
    "accepts_new_authorizations": true
  },
  {
    "service": "almond",
    "min_version": "0.102.0",
    "accepts_new_authorizations": true
  },
  {
    "service": "senz",
    "min_version": "2022.3.0",
    "accepts_new_authorizations": true
  },
  {
    "service": "bold",
    "min_version": "2022.10.0",
    "accepts_new_authorizations": true
  },
  {
    "service": "netatmo",
    "min_version": "0.105.0",
    "accepts_new_authorizations": true
  },
  {
    "service": "iotty",
    "min_version": "2024.8.0",
    "accepts_new_authorizations": true
  }
]
```

netatmo/test_init.py:from tests.components.cloud import mock_cloud
withings/test_init.py:from tests.components.cloud import mock_cloud


I think that might work

Reply:
Will need to do some more archeology

Reply:
I'd patch `async_provide_implementation` in `cloud/account_link.py` to provide a regular implementation, so that we can use the normal approach to testing config flow etc.

Reply:
Maybe we should simply patch `async_setup` in cloud so that nothing is setup and keep testing as is? Cloud is tested separately.

Reply:
Sounds good I’ll tackle it after I get some lunch